app/script/components/groupVideoGrid.js

```diff
@@ -22,189 +22,131 @@
-        [SIZES.FOURTH_SCREEN]: '',
-      };
+    const isAlone = grid.every((value, i) => i === index || value === 0);
+    const hasVerticalNeighbor = index % 2 === 0 ? grid[index + 1] !== 0 : grid[index - 1] !== 0;
 
-      const isSelfVideo = this.grid()[index] === this.selfId();
-      const mirrorClass = isSelfVideo && this.mirrorSelf ? ' mirror' : '';
-      return `group-video-grid__element${index} ${extraClasses[size]} ${mirrorClass}`;
+    if (isAlone) {
+      return SIZES.FULL_SCREEN;
+    } else if (!hasVerticalNeighbor) {
+      return SIZES.HALF_SCREEN;
     }
+    return SIZES.FOURTH_SCREEN;
+  }
 
-    getUIEValueForVideo(index) {
-      const size = this.getSizeForVideo(index);
-      const SIZES = GroupVideoGrid.CONFIG.VIDEO_ELEMENT_SIZE;
-      const extraClasses = {
-        [SIZES.EMPTY]: '',
-        [SIZES.FULL_SCREEN]: 'full',
-        [SIZES.HALF_SCREEN]: 'half',
-        [SIZES.FOURTH_SCREEN]: 'fourth',
-      };
-      return extraClasses[size];
-    }
+  getClassNameForVideo(index) {
+    const size = this.getSizeForVideo(index);
+    const SIZES = GroupVideoGrid.CONFIG.VIDEO_ELEMENT_SIZE;
+    const extraClasses = {
+      [SIZES.EMPTY]: 'group-video-grid__element--empty',
+      [SIZES.FULL_SCREEN]: 'group-video-grid__element--full-size',
+      [SIZES.HALF_SCREEN]: 'group-video-grid__element--full-height',
+      [SIZES.FOURTH_SCREEN]: '',
+    };
+
+    const isSelfVideo = this.grid()[index] === this.selfId();
+    const mirrorClass = isSelfVideo && this.mirrorSelf ? ' mirror' : '';
+    return `group-video-grid__element${index} ${extraClasses[size]} ${mirrorClass}`;
+  }
+
+  getUIEValueForVideo(index) {
+    const size = this.getSizeForVideo(index);
+    const SIZES = GroupVideoGrid.CONFIG.VIDEO_ELEMENT_SIZE;
+    const extraClasses = {
+      [SIZES.EMPTY]: '',
+      [SIZES.FULL_SCREEN]: 'full',
+      [SIZES.HALF_SCREEN]: 'half',
+      [SIZES.FOURTH_SCREEN]: 'fourth',
+    };
+    return extraClasses[size];
   }
+};
 
-  ko.components.register('group-video-grid', {
-    template: `
+ko.components.register('group-video-grid', {
+  template: `
       <div class="group-video" data-bind="template: {afterRender: scaleVideos}">
         <div class="group-video-grid" data-bind="foreach: {data: grid, as: 'streamId'}">
           <!-- ko if: streamId !== 0 -->
             <div class="group-video-grid__element" data-bind="css: $parent.getClassNameForVideo($index()), attr: {'data-uie-name': 'item-grid', 'data-uie-value': $parent.getUIEValueForVideo($index())}">
               <video autoplay class="group-video-grid__element-video" data-bind="sourceStream: $parent.getParticipantStream(streamId), muteMediaElement: $parent.getParticipantStream(streamId)">
               </video>
-              <!-- ko if: streamId === $parent.selfId() && !$parent.selfStreamState.audioSend() -->
+              <!-- ko if: streamId === $parent.selfId() && $parent.selfStreamMuted -->
```

Comment:
Isn't this an observable? I think in this case with the concatenated ko if you explicitly have to add (). At least it won't hurt to be more specific.